aws/cloudformation/bootstrap_chef_stack.sh.erb

```diff
@@ -76,14 +76,21 @@ CHAIN=$(aws s3 cp $CURRENT/chain.pem - | replace_newline)
 KEY=$(aws s3 cp $CURRENT/key.pem - | replace_newline)
 <% end -%>
 
-<% if database && !rack_env?(:production) -%>
 apt-get install -y  jq
+
+<% if database && !rack_env?(:production) -%>
 DB_SECRET_NAME=${DatabaseSecretAdmin}
 DB_SECRET=$(aws --region $REGION secretsmanager get-secret-value --secret-id $DB_SECRET_NAME --version-stage AWSCURRENT | jq -r ".SecretString")
 DB_USERNAME=$(echo $DB_SECRET | jq -r ".username")
 DB_PASSWORD=$(echo $DB_SECRET | jq -r ".password")
 <% end -%>
 
+<% if rack_env?(:adhoc) -%>
+# Chef managed environments (staging, test, levelbuilder, production) authenticate to GitHub with an SSH key.
+# We're not using curly braces to expand ENVIRONMENT for complex reasons (this script is processed by CloudFormation !Sub).
```

Comment:
nit; please expand on "for complex reasons." The purpose of comments is to be the place where we explain complex reasons so that future maintainers can know what they need to pay attention to when making refactoring decisions.

Reply:
I've fixed the complex interplay of ERB, bash string interpolation, and CloudFormation `!Sub` string interpolation, so this comment isn't needed anymore and we now use consistent syntax throughout the script.